### PR TITLE
Add helper function for mapping between action spaces

### PR DESCRIPTION
**Pull request by erikwijmans**
## Motivation and Context

We recently re-order our action number, which breaks checkpoints :(  This PR adds a helper function to help with mapping from the old action space to the new action space.  The help function maps on the level of probabilities or logits as this is more general and differentiable, making it also compatible with imitation learning.

## How Has This Been Tested

Via tests and applying this to our depth-PPO baselines.  On master, that baselines gets an SPL of 0, using this mapping, it gets 0.58 SPL (which is what it gets on the leaderboard).

## Types of changes

- New feature (non-breaking change which adds functionality)

### CODE REVIEW

File: test/test_action_mapping.py [hunk 1]
```diff
@@ -0,0 +1,86 @@
+import numpy as np
```
Comment (mathfac) [inline test/test_action_mapping.py]: Facebook header is missing.

File: habitat/utils/action_space_mapping.py [hunk 2]
```diff
@@ -0,0 +1,121 @@
+from typing import Dict
```
Comment (mathfac) [inline habitat/utils/action_space_mapping.py]: Facebook header is missing.

File: habitat/utils/action_space_mapping.py [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+        mask = np.ones_like(logits) @ mapping_matrix
+        flmin = np.full(
+            (logits.shape[0], mask.shape[1]), np.finfo(np.float32).min
+        )
+
+    return logits @ mapping_matrix + (1.0 - mask) * flmin
+
+
+def map_action_distribution(
+    mapping: Dict[int, int], *, probs=None, logits=None, use_torch=False
```
Comment (mathfac) [inline habitat/utils/action_space_mapping.py]: Can we add types for `probs , logits`?

**Comment by mathfac** [comment]
Feel free to reopen the PR if it's relevant. Closing for now.
